WalletWasabi.Gui/Controls/WalletExplorer/SendTabViewModel.cs

```diff
@@ -80,9 +82,25 @@ public string PayjoinEndPoint
 
 		private IPayjoinClient GetPayjoinClient()
 		{
-			if (!string.IsNullOrWhiteSpace(PayjoinEndPoint) && Uri.IsWellFormedUriString(PayjoinEndPoint, UriKind.Absolute))
+			if (!string.IsNullOrWhiteSpace(PayjoinEndPoint) &&
+				Uri.IsWellFormedUriString(PayjoinEndPoint, UriKind.Absolute))
 			{
 				var payjoinEndPointUri = new Uri(PayjoinEndPoint);
+				if (!Global.Config.UseTor)
+				{
+					if (payjoinEndPointUri.DnsSafeHost.EndsWith(".onion", StringComparison.OrdinalIgnoreCase))
+					{
+						Logger.LogWarning("Payjoin server is a hidden service but Tor is disabled. Ignoring...");
+						return null;
+					}
+
+					if (Global.Config.Network != Network.RegTest && payjoinEndPointUri.Scheme != "https")
```

Comment:
This should be only required for Main network.

```suggestion
					if (Global.Config.Network == Network.Main && payjoinEndPointUri.Scheme != Uri.UriSchemeHttps)
```